docs/reference/providers/faas-providers.md

```diff
@@ -0,0 +1,25 @@
+# `faas-provider` Interface
+The `faas-provider` provides the CRUD API for Functions as well as the invoke capabilities.
+
+The OpenFaaS team builds and uses the [`faas-provider`](https://github.com/openfaas/faas-provider) project to define the core interface and standard utilities of an OpenFaaS provider.  This interface, written in Go, ensures that you can create a provider which is compliant and compatible with the OpenFaaS gateway.
```

Comment:
The page is quite sparse, so are there any notes or points that you'd like to include from my blog post? https://blog.alexellis.io/faas-containerd-serverless-without-kubernetes/ or https://blog.alexellis.io/the-power-of-interfaces-openfaas/

Reply:
I think the technical details about how to write a provider make more sense in the provider repo. The vast majority of OpenFaaS users will be using existing providers, not writing new ones.  Adding a lot of documentation seems like it will cause a lot of difficulty in the future, e.g. keeping it in sync with the faas-provider repo and it's own godocs. I am wary of writing duplicate documentation.

Reply:
There is not enough context for my liking, what can you add that doesn't duplicate?

Perhaps some bullets that cover the main responsibilities of a provider?

Function CRUD
Logs streaming
Secret CRUD

Anything else?

Reply:
I'll make this change.